graph-proxy/src/graphql/workflows.rs

```diff
@@ -52,13 +64,18 @@ impl Workflow {
     pub(super) fn new(
         value: IoArgoprojWorkflowV1alpha1Workflow,
         visit: Visit,
+        artifacts: Vec<Artifact>,
```

Comment:
Since `artifacts` are derived from the `value.status`, we should compute them in this constructor instead of the resolver - as is done with the `WorkflowStatus` struct

Reply:
Done the first bullet point in c79eb68

Reply:
I'd rather we did the reorganisation now rather than being in this in-between state, happy to sit down an peer-program it if you'd fancy that